extensions/amp-list/0.1/amp-list.js

```diff
@@ -218,6 +218,13 @@ export class AmpList extends AMP.BaseElement {
       this.element.setAttribute('aria-live', 'polite');
     }
 
+    // for amp-lists with 'single-item' enabled, we should not specify the
+    // role of 'list' on the container as it causes screen readers to
+    // read an extra nested list with one item
+    if (this.element.hasAttribute('single-item')) {
+      this.container_.removeAttribute('role');
```

Comment:
Should this maybe be part of `createContainer_`?

Reply:
Thanks for the review, I also considered putting it there, but saw that the buildCallback seemed to be where all the rest of the attribute checks were so wanted to keep everything together.  I can change if you feel strongly about it.

Reply:
I think the concern fits better with `createContainer_`. The flow is somewhat confusing when adding the attribute, appending the child, and removing it later.

It would be cleaner to just not add it in that case, plus you could consolidate this one special case in one utility:

```js

  createContainer_() {
    const container = this.win.document.createElement('div');
    this.maybeSetRole_(container, 'list');
    // ...
    return container;
  }

  addElementsToContainer_(elements, container) {
    elements.forEach(element => {
      if (!element.hasAttribute('role')) {
        this.maybeSetRole_(element, 'listitem');
        // ... 
      }
    });
  }

  maybeSetRole_(element, role) {
    // With single-item, role=list|listitem is awkward for screen readers
    // since we don't need the extra nested list step.
    if (this.element.hasAttribute('single-item')) {
      return;
    }
    element.setAttribute('role', role);
  }
```

Reply:
Thanks for the recommendations Alan.  I've implemented your recommendations, it looks much cleaner and more efficient for bundle size.